torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -704,11 +712,12 @@ void SegmentedFusion::finalize() {
   // Keep track of groups that need to update expr list,
   //  including both the producer and consumer of the selected tv's that
   //  we cast to fp16.
-  std::unordered_set<SegmentedGroup*> affected_group_set;
-
+  std::unordered_set<SegmentedGroup*> affected_group_setfp16;
+  std::unordered_set<SegmentedGroup*> affected_group_setbf16;
```

Comment:
Thanks for simplifying the implementation. I think here is a bit more duplication to clean up :). Might have been rebasing issue again.

Reply:
addressed, sorry for the mixup. Let me know if there are any additional issues